react-front-end/tsrc/modules/LegacySelectionSessionModule.ts

```diff
@@ -469,29 +473,30 @@ export const selectResource = (
   itemKey: string,
   attachments: string[]
 ): Promise<void> =>
-  match(
-    [
-      Literal("coursesearch"),
-      () => selectResourceForCourseList(itemKey, attachments),
-    ],
-    [
-      Literal("search"),
-      () => selectResourceForSelectOrAdd(itemKey, attachments),
-    ],
-    [
-      Literal("skinnysearch"),
-      () => {
-        // Each selection in Skinny can only have one attachment.
-        if (attachments.length <= 1) {
-          const attachmentUUID =
-            attachments.length === 1 ? attachments[0] : undefined;
-          return selectResourceForSkinny(itemKey, attachmentUUID);
-        }
-        return Promise.reject(
-          new Error(
-            "Only one attachment is allowed in Skinny Selection Session."
+  pipe(
+    getSelectionSessionInfo().layout,
+    simpleMatch({
+      coursesearch: () => selectResourceForCourseList(itemKey, attachments),
+      search: () => selectResourceForSelectOrAdd(itemKey, attachments),
+      skinnysearch: () =>
+        pipe(
+          attachments,
+          E.fromPredicate<Error, string[]>(
+            (xs: string[]) => xs.length <= 1, // Each selection in Skinny can only have one attachment.
+            () =>
+              new Error(
+                "Only one attachment is allowed in Skinny Selection Session."
+              )
+          ),
+          E.map(A.head),
+          E.match(
+            (e) => Promise.reject(e),
+            (attachmentUUID) =>
```

Comment:
```suggestion
            (attachmentUUID: O.Option<string>) =>
```

At one point I thought the logic would have been changed, but then I quickly realised this is an option now. So all good.

Reply:
Yeah, I was wondering about doing something more there. Rather than just adding the type I figured I'd highlight what is going on here - as it does border on odd:

```typescript
          E.match(
            (e) => Promise.reject(e),
            flow(
              // If the array was empty, then pass `undefined` to `selectResourceForSkinny` which
              // indicates no attachment is selected - just the item.
              O.toUndefined,
              (attachmentUUID) =>
                selectResourceForSkinny(itemKey, attachmentUUID)
            )
          )
```

It's odd there's so much logic in this block for skinny selection sessions when compared to the others. But extracting in this PR is probably going to far.

Reply:
This comment helps a lot!